tests/spec_decode/test_multi_step_worker.py

```diff
@@ -440,3 +564,116 @@ def test_draft_proposals_mixed_k():
     assert proposals.proposal_lens.tolist() == [
         k for _ in range(expected_num_proposal_seqs - 1)
     ] + [0 for _ in range(expected_num_no_proposal_seqs)] + [k]
+
+
+@torch.inference_mode()
+def test_expand_execute_model_request_for_bonus_tokens():
```

Comment:
Why do we need both this test and `test_same_output_for_multi_step_with_batch_expansion`?

Reply:
fine to have both but we should add something in the docstring which explains the difference

Reply:
Removed this test as well as test_filter_model_output. Both should be covered by test_same_output_for_multi_step_with_batch_expansion as you suggested.